### PR TITLE
fix: Avoid reporting warning the export when src module format is none esm

### LINKED ISSUE #2111: [Bug Report]: Avoid reporting warning the export when src module format is none esm
### System Info

not related to system

### Details

From https://github.com/web-infra-dev/rspack/discussions/2110

### Reproduce link

_No response_

### Reproduce Steps

See https://github.com/web-infra-dev/rspack/discussions/2110

### PR DESCRIPTION
**Pull request by IWANABETHATGUY**
## Related issue (if exists)
Fixed #2111

## Types of changes

- [ ] Docs change / Dependency upgrade
- [ ] Bug fix
- [ ] New feature / Improvement
- [ ] Refactoring
- [ ] Breaking change

## Checklist

- [ ] I have added changeset via `pnpm run changeset`.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by IWANABETHATGUY** [comment]
!bench

**Comment by IWANABETHATGUY** [comment]
!bench

File: crates/rspack_error/tests/fixtures.rs [hunk 1]
```diff
@@ -25,7 +25,7 @@ pub async fn test_fixture<F: FnOnce(&Stats, Settings) -> rspack_error::Result<()
   f(&stats, settings)
 }
 
-#[fixture("tests/fixtures/*")]
+#[fixture("tests/fixtures/*", exclude("export_star_error"))]
```
Comment (IWANABETHATGUY) [inline crates/rspack_error/tests/fixtures.rs:28]: I am using explicit filter, instead of `.` prefix.

**Review by h-a-n-a — APPROVED** [review]
Looks good, please add a changeset before merging

**Review by hyf0 — APPROVED** [review]

**Comment by IWANABETHATGUY** [comment]
@h-a-n-a @hyf0 I have been added the changeset, would you mind approving again?

**Review by hyf0 — APPROVED** [review]

**Pull request merged**